### PR TITLE
add support for kts-based buildscripts

### LINKED ISSUE #14: generated module fails to build
Creating a module from the generator-cli results in a project that fails to build for two reasons:

1. The module plugin that is applied does not specify the version.  

Instead of:

```
plugins {
    id("io.ia.sdk.modl")
}
```

It should emit:

```
plugins {
    id("io.ia.sdk.modl") version("<some version>")
}
```

In addition, the module-signer fails to to resolve due to missing the repository in the settings.gradle pluginManagement.  At a minimum, we should add the following:

```
 maven { url = uri("https://nexus.inductiveautomation.com/repository/inductiveautomation-releases/") }
```

### LINKED ISSUE #24: Finish implementation of module generator support for build.gradle.kts files
Much of this work is already done, but we need to finish adding support to gain fully functional build.gradle.kts projects out of the module generator core.  When we do that, we should also add the kts support to the CLI.

- [ ] Finish templates and support for Kotlin based projects
- [ ] Finish support Kotlin for based buildscript templates

### PR DESCRIPTION
**Pull request by PerryAJ**
These changes do some cleanup, and add support for generating functional module structures using kotlin-based buildscripts.

Additions:

1. generate functional modules with kotlin-based buildscripts
2. updated gradle versions and various dependencies across the board, defaulting to 7.5.1 for new modules
3. added kotlin toolchain configuration, updated to target jvm 11
4. added some commentary to the generated module files to inform users that the generated project is simply a starting point
5. added integration tests to the generator-core that actually assembles a variety of modules and then executes a build against them
6.  updated version of ktlint, and made some filename changes to follow more recent best practices enforced by the linter.

Changes:

1. kotlin-based buildscripts are now the default for the generator
2. added some additional TemplateMarkers to better support kts buildscripts and customization

Fixes #14 
Fixes #24

### CODE REVIEW

File: generator/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+```
+
+```
+// windows
+gradlew.bat build
+```
+
+### Running the CLI
+
+To run the CLI locally, you from the `generator/` directory, you can run the `runCli --console plain` task, using the
```
Comment (paul-griffith) [inline generator/README.md:28]: Accidentally a word?

File: generator/build.gradle.kts [hunk 2]
```diff
@@ -16,9 +16,9 @@ buildscript {
 
 plugins {
     base
-    kotlin("jvm") version "1.5.21" apply false
-    id("org.jetbrains.dokka") version "1.5.0" apply false
-    id("com.diffplug.spotless") version "5.14.2" apply false
+    kotlin("jvm") version "1.6.21" apply false
```
Comment (paul-griffith) [inline generator/build.gradle.kts:19]: Curious why 1.6 over 1.7?

File: generator/generator-core/README.md [hunk 3]
```diff
@@ -8,72 +8,45 @@ We've had requests to make it easier to get started with module development usin
...
+can save a lot of time getting started.
 
 
-## Adding to Your Project
+## Adding to Your Project as a Dependency
 
-This small library is intended for use in a Java runtime environment, and is currently published to the Inductive
-Automation public repository. To use in a maven or gradle project, add the IA artifact repo to your project's
-repositories, then add the dependency.
+This small library is intended for use in a Java runtime environment, and is not currently to a public artifact
```
Comment (paul-griffith) [inline generator/generator-core/README.md:19]: `not currently to`?

File: generator/generator-core/build.gradle.kts [hunk 4]
```diff
@@ -19,29 +22,41 @@ java {
...
         }
     }
 
-    withType(JavaCompile::class) {
-        options.encoding = "UTF-8"
+    withType<Test>().configureEach {
+        // javaLauncher.set(javaToolchains.launcherFor {
+        //     languageVersion.set(JavaLanguageVersion.of(11))
+        // })
     }
```
Comment (paul-griffith) [inline generator/generator-core/build.gradle.kts:60]: Dead code?

File: generator/generator-core/src/main/kotlin/io/ia/ignition/module/generator/util/extensions.kt [hunk 5]
```diff
@@ -32,7 +37,48 @@ fun String.toPackagePath(maybeScope: ProjectScope? = null): String {
...
+                        if (line.contains(it.key)) {
+                            line = line.replace(it.key, it.value)
+                        }
+                    }
+                    outStream.write("$line\n".toByteArray(Charsets.UTF_8))
+                }
+            }
+        }
     }
 }
```
Comment (paul-griffith) [inline generator/generator-core/src/main/kotlin/io/ia/ignition/module/generator/util/extensions.kt:83]: Should be able to do this file IO cleaner

Comment (paul-griffith) [reply generator/generator-core/src/main/kotlin/io/ia/ignition/module/generator/util/extensions.kt:83]: E.G. if you drop the return value:
```kotlin
fun File.replacePlaceholders(replacements: Map<String, String> = emptyMap()) {
    return inputStream().writeToFileWithReplacements(this, replacements)
}

fun InputStream.writeToFileWithReplacements(file: File, replacements: Map<String, String> = emptyMap()) {
    file.bufferedWriter().use { writer ->
        this.bufferedReader().lineSequence().forEach { line ->
            writer.write(
                replacements.entries.fold(line) { currentLine, (key, value) ->
                    currentLine.replace(key, value)
                }
            )
            writer.newLine()
        }
    }
}
```

File: generator/build.gradle.kts (hunk 2 above)
Comment (PerryAJ) [reply generator/build.gradle.kts:19]: good question, think I tried going straight to 1.7 and ran into errors relating to build plugins and kotlin-test configuration.   Circled back after your comment and seem to have it working, so will push the 1.7 upgrade on this PR.

File: generator/generator-core/src/main/kotlin/io/ia/ignition/module/generator/util/extensions.kt (hunk 5 above)
Comment (PerryAJ) [reply generator/generator-core/src/main/kotlin/io/ia/ignition/module/generator/util/extensions.kt:83]: I copy-pasta'd your suggestion for this, and ended up causing a bunch of test failures.  Going to leave the more verbose version in there for now due to lack of time to and will circle back to it in the next round of cleanups.

**Review by paul-griffith — APPROVED** [review]

**Review by thirdgen88 — APPROVED** [review]

**Review by paul-griffith — APPROVED** [review]

**Pull request merged**